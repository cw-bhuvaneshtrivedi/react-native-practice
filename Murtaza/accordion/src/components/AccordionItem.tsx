import { StyleSheet, Text, View, Dimensions } from "react-native";
import React from "react";

interface AccordionItemProps {
  itemName: string;
}
const AccordionItem: React.FC<AccordionItemProps> = ({ itemName }) => {
  return (
    <View style={styles.container} testID="accordion-item">
      <View style={styles.item}>
        <Text style={styles.text} >{itemName}</Text>
      </View>
      <View style={styles.line}></View>
    </View>
  );
};

export default AccordionItem;

const styles = StyleSheet.create({
  container: {
    width: Dimensions.get("screen").width,
  },
  item: {
    backgroundColor: "#f9f9f9",
    paddingLeft: 36,
    paddingTop: 16,
    paddingBottom: 16,
    height: 50,
  },
  text: {
    fontSize: 15,
    color: "#484848",
  },
  line: {
    backgroundColor: "#efefef",
    width: "100%",
    height: 1,
  },
});
