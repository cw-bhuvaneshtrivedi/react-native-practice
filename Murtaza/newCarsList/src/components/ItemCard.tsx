import { StyleSheet, View, Text, TouchableOpacity } from "react-native";
import { FontAwesome } from "@expo/vector-icons";
import { useState } from "react";

export const ItemCard = ({ variations, select }: any) => {
  const [shortlist, setShortlist] = useState(false);
  const [compare, setCompare] = useState(false);
  const handleComparePress = () => {
    setCompare(!compare);
    select((count: number) => {
      return !compare ? count + 1 : count - 1;
    });
    //console.log("testing check")
  };
  const handleBookmark = () => {
    setShortlist(!shortlist);
    select((count: number) => {
      return !shortlist ? count + 1 : count - 1;
    });
  }
  return (
    <View style={style.container}>
      <View style={style.variationInfo}>
        <Text style={style.variationName} testID="carName">
          {variations.name}
        </Text>
        <View style={style.carDescriptionContainer}>
          <Text style={style.carDescriptionContainerText}>
            {variations.maxPower}
          </Text>
          <Text style={style.carDescriptionContainerText}>
            {variations.displacement}
          </Text>
          <Text style={style.carDescriptionContainerText}>
            {variations.transmission}
          </Text>
          <Text style={style.carDescriptionContainerText}>
            {variations.fuelType}
          </Text>
        </View>

        <View style={style.carPriceContainerMain}>
          <Text style={style.carPriceMainText}>
            {variations.priceOverview.price}
          </Text>
          <Text style={style.carPriceSecondaryText}>Show price in my city</Text>
          <Text style={style.carPriceEMIText}>EMI</Text>
        </View>
        <View style={style.carPriceContainerSecondary}>
          <Text style={style.carShowRoomPriceText}>Avg.Ex-Showroom Price</Text>
          <Text style={{ fontStyle: "italic" }}>{variations.emi}</Text>
        </View>
      </View>

      <View style={style.containerOptions}>
        <TouchableOpacity
          onPress={handleBookmark}
          style={{ justifyContent: "center" }}
          testID="shortlistOptionButton"
        >
          <View
            style={style.containerOptionsElementLeft}
            testID="shortlistOption"
          >
            {shortlist ? (
              <FontAwesome name="bookmark" size={15} color="#00afa0" />
            ) : (
              <FontAwesome name="bookmark-o" size={15} color="#6f6f6f" />
            )}
            {shortlist ? (
              <Text style={style.containerOptionsText}>Shortlisted</Text>
            ) : (
              <Text style={style.containerOptionsText}>Shortlist</Text>
            )}
          </View>
        </TouchableOpacity>
        <View style={style.containerOptionsElementCenter}></View>
        <TouchableOpacity
          onPress={handleComparePress}
          style={{ justifyContent: "center" }}
          testID="compareOptionButton"
        >
          <View
            style={style.containerOptionsElementRight}
            testID="compareOption"
          >
            <FontAwesome
              name="balance-scale"
              size={13}
              color={compare ? "#00afa0" : "#6f6f6f"}
            />
            {compare ? (
              <Text style={style.containerOptionsText}>
                Remove from Compare
              </Text>
            ) : (
              <Text style={style.containerOptionsText}>Add to Compare</Text>
            )}
          </View>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const style = StyleSheet.create({
  container: {
    marginRight: 14,
    width: 321,
    height: 170,
    backgroundColor: "white",
    borderRadius: 2,
    borderWidth: 2,
    borderColor: "#e1e1e1",
    justifyContent: "space-between",
  },
  variationInfo: {
    paddingLeft: 12,
    paddingRight: 12,
  },
  variationName: {
    paddingTop: 12,
    fontSize: 16,
    color: "#0288d1",
  },
  carShowRoomPriceText: {
    fontSize: 12,
    color: "#aaaaaa",
    alignSelf: "flex-end",
  },
  carEMIPrice: {
    color: "#484848",
    fontSize: 13,
  },
  carPriceMainText: {
    fontSize: 16,
    color: "#484848",
    fontWeight: "bold",
  },
  carPriceSecondaryText: {
    fontSize: 13,
    color: "#0288d1",
    alignSelf: "flex-end",
  },
  carPriceEMIText: {
    fontSize: 13,
    color: "#aaaaaa",
    alignSelf: "flex-end",
    fontStyle: "italic",
  },
  carPriceContainerMain: {
    paddingTop: 16,
    flexDirection: "row",
    justifyContent: "space-between",
  },
  carPriceContainerSecondary: {
    paddingTop: 4,
    flexDirection: "row",
    height: 20,
    justifyContent: "space-between",
  },
  carDescriptionContainer: {
    width: 233,
    paddingTop: 8,
    flexDirection: "row",
    justifyContent: "space-between",
    flexWrap: "wrap",
  },
  carDescriptionContainerText: {
    color: "#aaaaaa",
    fontSize: 13,
  },
  containerOptions: {
    flexDirection: "row",
    height: 40,
    backgroundColor: "#f9f9f9",
    borderRadius: 2,
    justifyContent: "space-between",
  },
  containerOptionsElementLeft: {
    width: 120,
    justifyContent: "center",
    alignItems: "center",
    flexDirection: "row",
  },
  containerOptionsElementRight: {
    width: 200,
    justifyContent: "center",
    flexDirection: "row",
    alignSelf: "center",
  },
  containerOptionsElementCenter: {
    alignSelf: "center",
    height: 20,
    width: 1,
    backgroundColor: "#e1e1e1",
  },
  containerOptionsText: {
    paddingLeft: 4,
    color: "#6f6f6f",
  },
});
