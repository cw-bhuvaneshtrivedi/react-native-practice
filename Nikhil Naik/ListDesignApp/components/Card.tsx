import { StyleSheet, Text, View } from "react-native";
import React from "react";

const Card = ({ data }) => {
  return (
    <View>
      <View style={styles.container}>
        <Text style={styles.containerText} testID="cardName">
          {data}
        </Text>
      </View>
      <View style={styles.seperator}></View>
    </View>
  );
};

export default Card;

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    justifyContent: "flex-start",
    paddingLeft: 38,
    backgroundColor: "#f9f9f9",
    width: "100%",
  },
  containerText: {
    color: "#484848",
    fontSize: 15,
    paddingTop: 16,
    paddingBottom: 16,
    // width: "50%",
  },
  seperator: {
    width: "100%",
    height: 2,
    backgroundColor: "#efefef",
  },
});
