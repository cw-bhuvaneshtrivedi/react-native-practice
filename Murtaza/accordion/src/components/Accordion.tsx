import { StyleSheet, Text, TouchableOpacity, View, Dimensions } from "react-native";
import React, { useState, useEffect } from "react";
import { Entypo } from "@expo/vector-icons";
import AccordionItem from "./AccordionItem";
import Animated, {
  useAnimatedStyle,
  useSharedValue,
  withTiming,
  Easing,
} from "react-native-reanimated";

interface AccordionProps {
  title: string;
  items: string[];
  isOpen: boolean;
  onPress: () => void;
}

const Accordion: React.FC<AccordionProps> = ({ title, items, isOpen, onPress }) => {
  const rotateZ = useSharedValue(isOpen ? -180 : 0);

  useEffect(() => {
    rotateZ.value = withTiming(isOpen ? -180 : 0, {
      duration: 350,
      easing: Easing.inOut(Easing.quad),
    });
  }, [isOpen]);

  const handleAccordionPress = () => {
    onPress();
  };

  const animatedStyle = useAnimatedStyle(() => ({
    transform: [{ rotateZ: `${rotateZ.value}` + "deg" }],
  }));

  return (
    <View>
      <TouchableOpacity style={styles.container} onPress={handleAccordionPress}>
        <Text style={styles.accordionTitle}>{title}</Text>
        <Animated.View style={animatedStyle}>
          <Entypo name="chevron-down" size={24} color="black" />
        </Animated.View>
      </TouchableOpacity>
      <View style={styles.line}></View>

      {isOpen && (
        <View>
          {items.map((item, index) => (
            <AccordionItem key={index} itemName={item} />
          ))}
        </View>
      )}
    </View>
  );
};

export default Accordion;

const styles = StyleSheet.create({
  container: {
    backgroundColor: "white",
    paddingTop: 16,
    paddingLeft: 28,
    paddingRight: 31,
    flexDirection: "row",
    justifyContent: "space-between",
    height: 50,
    // marginTop: 20,
  },
  accordionTitle: {
    color: "#484848",
    fontSize: 15,
  },
  line: {
    backgroundColor: "#efefef",
    width: "100%",
    height: 1,
  },
});
