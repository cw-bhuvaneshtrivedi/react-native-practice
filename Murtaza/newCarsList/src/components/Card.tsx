import { useState } from "react";
import { StyleSheet, View, Text, Image, Dimensions } from "react-native";
import { MaterialIcons } from "@expo/vector-icons";
import { FlashList } from "@shopify/flash-list";
import { ItemCard } from "./ItemCard";
export const Card = ({ item }:any) => {
  const [selected, SetSelected] = useState(0);
  return (
    <View style={styles.row}>
      <View
        style={{
          height: 305,
          width: 5,
          backgroundColor: selected > 0 ? "#00afa0" : "white",
        }}
        testID="sideLine"
      ></View>
      <View style={styles.card}>
        <View style={styles.head}>
          <Image
            source={{
              uri: item.hostUrl + "600x337" + item.originalImagePath,
            }}
            style={styles.image}
            testID="carImage"
          />
          <View style={styles.HeadText}>
            <Text style={styles.company} testID="makerName">
              {item.makeName}
            </Text>
            <Text style={styles.model} testID="modelName">
              {item.modelName}
            </Text>
          </View>
          <View style={styles.arrow}>
            <MaterialIcons name="arrow-forward-ios" size={15} color="gray" />
          </View>
        </View>
        <FlashList
          data={item.versions}
          renderItem={({ item }) => (
            <ItemCard variations={item} select={SetSelected} />
          )}
          horizontal={true}
          showsHorizontalScrollIndicator={false}
          estimatedItemSize={335}
          estimatedListSize={{
            height: 170,
            width: Dimensions.get("window").width,
          }}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  row: {
    flex: 1,
    flexDirection: "row",
  },
  card: {
    flex: 1,
    width: 360,
    paddingLeft: 10,
    paddingTop: 13,
    backgroundColor: "white",
    height: 305,
    justifyContent: "flex-end",
  },
  head: {
    width: 360,
    paddingBottom: 20,
    paddingTop: 20,
    paddingLeft: 7,
    flexDirection: "row",
    justifyContent: "flex-start",
  },
  image: {
    width: 130,
    height: 73,
  },
  HeadText: {
    paddingTop: 16,
    paddingBottom: 15,
    paddingLeft: 16,
  },
  company: {
    fontSize: 13,
    color: "#aaaaaa",
  },
  model: {
    fontSize: 18,
    color: "#484848",
    paddingTop: 4,
  },
  arrow: {
    marginLeft: 66,
    alignSelf: "center",
  },
});
