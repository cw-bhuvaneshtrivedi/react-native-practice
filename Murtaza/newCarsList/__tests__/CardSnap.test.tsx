import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { Card } from '../src/components/Card'; 

const mockItem = {
  makeName: 'Mock Make Name',
  modelName: 'Mock Model Name',
  hostUrl: 'https://mock.url/',
  originalImagePath: '/mock-image.jpg',
  versions: [
    {
      name: 'Mock Version Name',
      fuelType: 'Mock Fuel Type',
      maxPower: 'Mock Max Power',
      displacement: 'Mock Displacement',
      transmission: 'Mock Transmission',
      emi: 'Mock EMI',
      priceOverview: {
        price: 'Mock Price',
        label: 'Mock Label',
        reasonText: 'Mock Reason Text',
      },
    },
  ],
};

it('Card component renders correctly', () => {
  const tree = renderer.create(<Card item={mockItem} />).toJSON();
  expect(tree).toMatchSnapshot();
});
