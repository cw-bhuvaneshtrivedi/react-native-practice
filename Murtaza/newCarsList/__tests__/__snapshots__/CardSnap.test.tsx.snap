// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "white",
        "height": 305,
        "width": 5,
      }
    }
    testID="sideLine"
  />
  <View
    style={
      {
        "backgroundColor": "white",
        "flex": 1,
        "height": 305,
        "justifyContent": "flex-end",
        "paddingLeft": 10,
        "paddingTop": 13,
        "width": 360,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "paddingBottom": 20,
          "paddingLeft": 7,
          "paddingTop": 20,
          "width": 360,
        }
      }
    >
      <Image
        source={
          {
            "uri": "https://mock.url/600x337/mock-image.jpg",
          }
        }
        style={
          {
            "height": 73,
            "width": 130,
          }
        }
        testID="carImage"
      />
      <View
        style={
          {
            "paddingBottom": 15,
            "paddingLeft": 16,
            "paddingTop": 16,
          }
        }
      >
        <Text
          style={
            {
              "color": "#aaaaaa",
              "fontSize": 13,
            }
          }
          testID="makerName"
        >
          Mock Make Name
        </Text>
        <Text
          style={
            {
              "color": "#484848",
              "fontSize": 18,
              "paddingTop": 4,
            }
          }
          testID="modelName"
        >
          Mock Model Name
        </Text>
      </View>
      <View
        style={
          {
            "alignSelf": "center",
            "marginLeft": 66,
          }
        }
      >
        <Text />
      </View>
    </View>
    <View
      style={{}}
    >
      <RCTScrollView
        applyWindowCorrection={[Function]}
        canChangeSize={true}
        contentContainerStyle={
          {
            "backgroundColor": undefined,
            "minHeight": 1,
            "minWidth": 1,
            "paddingBottom": 0,
            "paddingTop": 0,
          }
        }
        contentHeight={170}
        contentWidth={335}
        data={
          [
            {
              "displacement": "Mock Displacement",
              "emi": "Mock EMI",
              "fuelType": "Mock Fuel Type",
              "maxPower": "Mock Max Power",
              "name": "Mock Version Name",
              "priceOverview": {
                "label": "Mock Label",
                "price": "Mock Price",
                "reasonText": "Mock Reason Text",
              },
              "transmission": "Mock Transmission",
            },
          ]
        }
        dataProvider={
          DataProvider {
            "_data": [
              {
                "displacement": "Mock Displacement",
                "emi": "Mock EMI",
                "fuelType": "Mock Fuel Type",
                "maxPower": "Mock Max Power",
                "name": "Mock Version Name",
                "priceOverview": {
                  "label": "Mock Label",
                  "price": "Mock Price",
                  "reasonText": "Mock Reason Text",
                },
                "transmission": "Mock Transmission",
              },
            ],
            "_firstIndexToProcess": 0,
            "_hasStableIds": false,
            "_requiresDataChangeHandling": false,
            "_size": 1,
            "getStableId": [Function],
            "rowHasChanged": [Function],
          }
        }
        disableRecycling={false}
        estimatedItemSize={335}
        extendedState={{}}
        externalScrollView={[Function]}
        finalRenderAheadOffset={250}
        forceNonDeterministicRendering={true}
        horizontal={true}
        initialOffset={0}
        initialRenderIndex={0}
        isHorizontal={true}
        layoutProvider={
          GridLayoutProviderWithProps {
            "_acceptableRelayoutDelta": 1,
            "_getHeightOrWidth": [Function],
            "_getLayoutTypeForIndex": [Function],
            "_getSpan": [Function],
            "_hasExpired": false,
            "_isHorizontal": true,
            "_lastLayoutManager": GridLayoutManager {
              "_acceptableRelayoutDelta": 1,
              "_getSpan": [Function],
              "_isGridHorizontal": true,
              "_isHorizontal": true,
              "_layoutProvider": [Circular],
              "_layouts": [
                {
                  "height": 170,
                  "type": 0,
                  "width": 335,
                  "x": 0,
                  "y": 0,
                },
              ],
              "_maxSpan": 1,
              "_renderWindowSize": {
                "height": 170,
                "width": 750,
              },
              "_totalHeight": 170,
              "_totalWidth": 335,
              "_window": {
                "height": 170,
                "width": 750,
              },
            },
            "_maxSpan": 1,
            "_renderWindowSize": {
              "height": 170,
              "width": 750,
            },
            "_setLayoutForType": [Function],
            "_tempDim": {
              "height": 0,
              "width": 0,
            },
            "averageWindow": AverageWindow {
              "currentAverage": 335,
              "currentCount": 1,
              "inputValues": [
                335,
                ,
                ,
                ,
                ,
                ,
              ],
              "nextIndex": 1,
            },
            "defaultEstimatedItemSize": 100,
            "layoutObject": {
              "size": undefined,
              "span": undefined,
            },
            "props": {
              "data": [
                {
                  "displacement": "Mock Displacement",
                  "emi": "Mock EMI",
                  "fuelType": "Mock Fuel Type",
                  "maxPower": "Mock Max Power",
                  "name": "Mock Version Name",
                  "priceOverview": {
                    "label": "Mock Label",
                    "price": "Mock Price",
                    "reasonText": "Mock Reason Text",
                  },
                  "transmission": "Mock Transmission",
                },
              ],
              "estimatedItemSize": 335,
              "estimatedListSize": {
                "height": 170,
                "width": 750,
              },
              "horizontal": true,
              "numColumns": 1,
              "renderItem": [Function],
              "showsHorizontalScrollIndicator": false,
            },
            "renderWindowInsets": {
              "height": 0,
              "width": 0,
            },
            "shouldRefreshWithAnchoring": true,
          }
        }
        layoutSize={
          {
            "height": 170,
            "width": 750,
          }
        }
        maxRenderAhead={750}
        numColumns={1}
        onEndReached={[Function]}
        onEndReachedThreshold={0}
        onEndReachedThresholdRelative={0}
        onItemLayout={[Function]}
        onLayout={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onSizeChanged={[Function]}
        onVisibleIndicesChanged={[Function]}
        removeClippedSubviews={false}
        renderAheadOffset={0}
        renderAheadStep={250}
        renderContentContainer={[Function]}
        renderItem={[Function]}
        renderItemContainer={[Function]}
        rowRenderer={[Function]}
        scrollEventThrottle={16}
        scrollThrottle={16}
        scrollViewProps={
          {
            "contentContainerStyle": {
              "backgroundColor": undefined,
              "minHeight": 1,
              "minWidth": 1,
              "paddingBottom": 0,
              "paddingTop": 0,
            },
            "onLayout": [Function],
            "onScrollBeginDrag": [Function],
            "refreshControl": undefined,
            "style": {
              "minHeight": 1,
              "minWidth": 1,
            },
          }
        }
        showsHorizontalScrollIndicator={false}
        style={
          {
            "minHeight": 1,
            "minWidth": 1,
          }
        }
        suppressBoundedSizeException={true}
        windowCorrectionConfig={
          {
            "applyToInitialOffset": true,
            "applyToItemScroll": true,
            "value": {
              "endCorrection": 0,
              "startCorrection": 0,
              "windowShift": -0,
            },
          }
        }
      >
        <View>
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                {
                  "paddingLeft": 0,
                  "paddingTop": 0,
                }
              }
            />
            <View
              style={
                [
                  undefined,
                  undefined,
                ]
              }
            />
            <AutoLayoutView
              enableInstrumentation={false}
              horizontal={true}
              onBlankAreaEvent={[Function]}
              onLayout={[Function]}
              renderAheadOffset={0}
              scrollOffset={0}
              style={
                {
                  "height": 170,
                  "width": 335,
                }
              }
              windowSize={750}
            >
              <CellContainer
                index={0}
                onLayout={[Function]}
                style={
                  {
                    "alignItems": "stretch",
                    "flexDirection": "row",
                    "height": 170,
                    "left": 0,
                    "position": "absolute",
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    {
                      "flexDirection": "column",
                    }
                  }
                >
                  <View
                    style={
                      {
                        "backgroundColor": "white",
                        "borderColor": "#e1e1e1",
                        "borderRadius": 2,
                        "borderWidth": 2,
                        "height": 170,
                        "justifyContent": "space-between",
                        "marginRight": 14,
                        "width": 321,
                      }
                    }
                  >
                    <View
                      style={
                        {
                          "paddingLeft": 12,
                          "paddingRight": 12,
                        }
                      }
                    >
                      <Text
                        style={
                          {
                            "color": "#0288d1",
                            "fontSize": 16,
                            "paddingTop": 12,
                          }
                        }
                        testID="carName"
                      >
                        Mock Version Name
                      </Text>
                      <View
                        style={
                          {
                            "flexDirection": "row",
                            "flexWrap": "wrap",
                            "justifyContent": "space-between",
                            "paddingTop": 8,
                            "width": 233,
                          }
                        }
                      >
                        <Text
                          style={
                            {
                              "color": "#aaaaaa",
                              "fontSize": 13,
                            }
                          }
                        >
                          Mock Max Power
                        </Text>
                        <Text
                          style={
                            {
                              "color": "#aaaaaa",
                              "fontSize": 13,
                            }
                          }
                        >
                          Mock Displacement
                        </Text>
                        <Text
                          style={
                            {
                              "color": "#aaaaaa",
                              "fontSize": 13,
                            }
                          }
                        >
                          Mock Transmission
                        </Text>
                        <Text
                          style={
                            {
                              "color": "#aaaaaa",
                              "fontSize": 13,
                            }
                          }
                        >
                          Mock Fuel Type
                        </Text>
                      </View>
                      <View>
                        <View
                          style={
                            {
                              "flexDirection": "row",
                              "justifyContent": "space-between",
                              "paddingTop": 16,
                            }
                          }
                        >
                          <Text
                            style={
                              {
                                "color": "#484848",
                                "fontSize": 16,
                                "fontWeight": "bold",
                              }
                            }
                          >
                            Mock Price
                          </Text>
                          <Text
                            style={
                              {
                                "alignSelf": "flex-end",
                                "color": "#0288d1",
                                "fontSize": 13,
                              }
                            }
                          >
                            Show price in my city
                          </Text>
                          <Text
                            style={
                              {
                                "alignSelf": "flex-end",
                                "color": "#aaaaaa",
                                "fontSize": 13,
                                "fontStyle": "italic",
                              }
                            }
                          >
                            EMI
                          </Text>
                        </View>
                        <View
                          style={
                            {
                              "flexDirection": "row",
                              "height": 20,
                              "justifyContent": "space-between",
                              "paddingTop": 4,
                            }
                          }
                        >
                          <Text
                            style={
                              {
                                "alignSelf": "flex-end",
                                "color": "#aaaaaa",
                                "fontSize": 12,
                              }
                            }
                          >
                            Avg.Ex-Showroom Price
                          </Text>
                          <Text
                            style={
                              {
                                "fontStyle": "italic",
                              }
                            }
                          >
                            Mock EMI
                          </Text>
                        </View>
                      </View>
                    </View>
                    <View
                      style={
                        {
                          "backgroundColor": "#f9f9f9",
                          "borderRadius": 2,
                          "flexDirection": "row",
                          "height": 40,
                          "justifyContent": "space-between",
                        }
                      }
                    >
                      <View
                        accessibilityState={
                          {
                            "busy": undefined,
                            "checked": undefined,
                            "disabled": undefined,
                            "expanded": undefined,
                            "selected": undefined,
                          }
                        }
                        accessibilityValue={
                          {
                            "max": undefined,
                            "min": undefined,
                            "now": undefined,
                            "text": undefined,
                          }
                        }
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          {
                            "justifyContent": "center",
                            "opacity": 1,
                          }
                        }
                        testID="shortlistOptionButton"
                      >
                        <View
                          style={
                            {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "justifyContent": "center",
                              "width": 120,
                            }
                          }
                          testID="shortlistOption"
                        >
                          <Text />
                          <Text
                            style={
                              {
                                "color": "#6f6f6f",
                                "paddingLeft": 4,
                              }
                            }
                          >
                            Shortlist
                          </Text>
                        </View>
                      </View>
                      <View
                        style={
                          {
                            "alignSelf": "center",
                            "backgroundColor": "#e1e1e1",
                            "height": 20,
                            "width": 1,
                          }
                        }
                      />
                      <View
                        accessibilityState={
                          {
                            "busy": undefined,
                            "checked": undefined,
                            "disabled": undefined,
                            "expanded": undefined,
                            "selected": undefined,
                          }
                        }
                        accessibilityValue={
                          {
                            "max": undefined,
                            "min": undefined,
                            "now": undefined,
                            "text": undefined,
                          }
                        }
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          {
                            "justifyContent": "center",
                            "opacity": 1,
                          }
                        }
                        testID="compareOptionButton"
                      >
                        <View
                          style={
                            {
                              "alignSelf": "center",
                              "flexDirection": "row",
                              "justifyContent": "center",
                              "width": 200,
                            }
                          }
                          testID="compareOption"
                        >
                          <Text />
                          <Text
                            style={
                              {
                                "color": "#6f6f6f",
                                "paddingLeft": 4,
                              }
                            }
                          >
                            Add to Compare
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </CellContainer>
            </AutoLayoutView>
            <CellContainer
              index={-1}
              style={
                [
                  undefined,
                  undefined,
                ]
              }
            />
            <View
              style={
                {
                  "paddingBottom": 0,
                  "paddingRight": 0,
                }
              }
            />
            <View
              pointerEvents="none"
              style={
                {
                  "opacity": 0,
                }
              }
            >
              <View
                style={
                  {
                    "backgroundColor": "white",
                    "borderColor": "#e1e1e1",
                    "borderRadius": 2,
                    "borderWidth": 2,
                    "height": 170,
                    "justifyContent": "space-between",
                    "marginRight": 14,
                    "width": 321,
                  }
                }
              >
                <View
                  style={
                    {
                      "paddingLeft": 12,
                      "paddingRight": 12,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#0288d1",
                        "fontSize": 16,
                        "paddingTop": 12,
                      }
                    }
                    testID="carName"
                  >
                    Mock Version Name
                  </Text>
                  <View
                    style={
                      {
                        "flexDirection": "row",
                        "flexWrap": "wrap",
                        "justifyContent": "space-between",
                        "paddingTop": 8,
                        "width": 233,
                      }
                    }
                  >
                    <Text
                      style={
                        {
                          "color": "#aaaaaa",
                          "fontSize": 13,
                        }
                      }
                    >
                      Mock Max Power
                    </Text>
                    <Text
                      style={
                        {
                          "color": "#aaaaaa",
                          "fontSize": 13,
                        }
                      }
                    >
                      Mock Displacement
                    </Text>
                    <Text
                      style={
                        {
                          "color": "#aaaaaa",
                          "fontSize": 13,
                        }
                      }
                    >
                      Mock Transmission
                    </Text>
                    <Text
                      style={
                        {
                          "color": "#aaaaaa",
                          "fontSize": 13,
                        }
                      }
                    >
                      Mock Fuel Type
                    </Text>
                  </View>
                  <View>
                    <View
                      style={
                        {
                          "flexDirection": "row",
                          "justifyContent": "space-between",
                          "paddingTop": 16,
                        }
                      }
                    >
                      <Text
                        style={
                          {
                            "color": "#484848",
                            "fontSize": 16,
                            "fontWeight": "bold",
                          }
                        }
                      >
                        Mock Price
                      </Text>
                      <Text
                        style={
                          {
                            "alignSelf": "flex-end",
                            "color": "#0288d1",
                            "fontSize": 13,
                          }
                        }
                      >
                        Show price in my city
                      </Text>
                      <Text
                        style={
                          {
                            "alignSelf": "flex-end",
                            "color": "#aaaaaa",
                            "fontSize": 13,
                            "fontStyle": "italic",
                          }
                        }
                      >
                        EMI
                      </Text>
                    </View>
                    <View
                      style={
                        {
                          "flexDirection": "row",
                          "height": 20,
                          "justifyContent": "space-between",
                          "paddingTop": 4,
                        }
                      }
                    >
                      <Text
                        style={
                          {
                            "alignSelf": "flex-end",
                            "color": "#aaaaaa",
                            "fontSize": 12,
                          }
                        }
                      >
                        Avg.Ex-Showroom Price
                      </Text>
                      <Text
                        style={
                          {
                            "fontStyle": "italic",
                          }
                        }
                      >
                        Mock EMI
                      </Text>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    {
                      "backgroundColor": "#f9f9f9",
                      "borderRadius": 2,
                      "flexDirection": "row",
                      "height": 40,
                      "justifyContent": "space-between",
                    }
                  }
                >
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "justifyContent": "center",
                        "opacity": 1,
                      }
                    }
                    testID="shortlistOptionButton"
                  >
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "justifyContent": "center",
                          "width": 120,
                        }
                      }
                      testID="shortlistOption"
                    >
                      <Text />
                      <Text
                        style={
                          {
                            "color": "#6f6f6f",
                            "paddingLeft": 4,
                          }
                        }
                      >
                        Shortlist
                      </Text>
                    </View>
                  </View>
                  <View
                    style={
                      {
                        "alignSelf": "center",
                        "backgroundColor": "#e1e1e1",
                        "height": 20,
                        "width": 1,
                      }
                    }
                  />
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "justifyContent": "center",
                        "opacity": 1,
                      }
                    }
                    testID="compareOptionButton"
                  >
                    <View
                      style={
                        {
                          "alignSelf": "center",
                          "flexDirection": "row",
                          "justifyContent": "center",
                          "width": 200,
                        }
                      }
                      testID="compareOption"
                    >
                      <Text />
                      <Text
                        style={
                          {
                            "color": "#6f6f6f",
                            "paddingLeft": 4,
                          }
                        }
                      >
                        Add to Compare
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
